go/mysql/server.go

```diff
@@ -134,6 +134,12 @@ type Handler interface {
 	// and the Vitess layer needs to parse the query to identify the query parameters so that the correct response
 	// packets can be sent.
 	ParserOptionsForConnection(c *Conn) (sqlparser.ParserOptions, error)
+
+	// ComRegisterReplica is called when a connection receives a ComRegisterReplica request
+	ComRegisterReplica(c *Conn, replicaHost string, replicaPort uint16, replicaUser string, replicaPassword string) error
```

Comment:
I would consider putting these in an extension to the main handler interface, e.g. `BinLogReplicaHandler`

Most people implementing these interfaces aren't going to want this capability (including Doltgres e.g.)